### PR TITLE
Make ClusterNodeState consistent store entries have a TTL

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/log/consistent_store-inl.h [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#ifndef CERT_TRANS_LOG_CONSISTENT_STORE_INL_H_
```
Comment (pphaneuf) [inline cpp/log/consistent_store-inl.h]: AAAARRRGGGHH!!!1!

File: cpp/log/etcd_consistent_store-inl.h [hunk 2]
```diff
@@ -267,7 +270,8 @@ util::Status EtcdConsistentStore<Logged>::SetClusterNodeState(
   // nobody else is updating our cluster state.
   // TODO(alcutter): These files should have an associated TTL.
```
Comment (pphaneuf) [inline cpp/log/etcd_consistent_store-inl.h]: Isn't that TODO addressed now?

Comment (AlCutter) [inline cpp/log/etcd_consistent_store-inl.h]: TODONE!

File: cpp/log/etcd_consistent_store.h [hunk 3]
```diff
@@ -71,6 +71,11 @@ class EtcdConsistentStore : public ConsistentStore<Logged> {
   template <class T>
   util::Status ForceSetEntry(const std::string& path, EntryHandle<T>* entry);
 
+  template <class T>
+  util::Status ForceSetEntryWithTTL(const std::string& path,
+                                    const std::chrono::duration<int>& ttl,
```
Comment (pphaneuf) [inline cpp/log/etcd_consistent_store.h]: I'd prefer using the ```std::chrono::seconds``` typedef, doesn't really change anything, but I find it nicer to read.

File: cpp/log/etcd_consistent_store-inl.h (hunk 2 above)
Comment (pphaneuf) [inline cpp/log/etcd_consistent_store-inl.h]: LOL!

File: cpp/log/etcd_consistent_store.h (hunk 3 above)
Comment (AlCutter) [inline cpp/log/etcd_consistent_store.h]: Fair enough, done.

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**